modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/appcatalog/StorageResourceRepository.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.airavata.registry.core.repositories.appcatalog;
+
+import org.apache.airavata.model.appcatalog.storageresource.StorageResourceDescription;
+import org.apache.airavata.model.commons.airavata_commonsConstants;
+import org.apache.airavata.registry.core.entities.appcatalog.StorageInterfaceEntity;
+import org.apache.airavata.registry.core.entities.appcatalog.StorageInterfacePK;
+import org.apache.airavata.registry.core.entities.appcatalog.StorageResourceEntity;
+import org.apache.airavata.registry.core.utils.AppCatalogUtils;
+import org.apache.airavata.registry.core.utils.DBConstants;
+import org.apache.airavata.registry.core.utils.QueryConstants;
+import org.apache.airavata.registry.cpi.AppCatalogException;
+import org.apache.airavata.registry.cpi.StorageResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by skariyat on 3/12/18.
+ */
+public class StorageResourceRepository extends AppCatAbstractRepository<StorageResourceDescription, StorageResourceEntity, String> implements StorageResource {
+
+    private final static Logger logger = LoggerFactory.getLogger(StorageResourceRepository.class);
+
+
+    public StorageResourceRepository() {
+        super(StorageResourceDescription.class, StorageResourceEntity.class);
+    }
+
+
+    @Override
+    public String addStorageResource(StorageResourceDescription description) throws AppCatalogException {
+        try {
+            final String storageResourceId = AppCatalogUtils.getID(description.getHostName());
+            if (description.getStorageResourceDescription().equals("") || description.getStorageResourceId().equals(airavata_commonsConstants.DEFAULT_ID)) {
```

Comment:
Use "".equals(description.getStorageResourceDescription()) and airavata_commonsConstants.DEFAULT_ID.equals(description.getStorageResourceId()) to avoid possible NPE